src/main/scala/org/constellation/ConfigUtil.scala

```diff
@@ -59,4 +61,29 @@ object ConfigUtil {
 
   def isEnabledCloudStorage: Boolean =
     isEnabledGCPStorage || isEnabledAWSStorage
+
+  case class AWSStorageConfig(accessKey: String, secretKey: String, region: String, bucket: String)
+
+  def loadAWSStorageConfigs(constellationConfig: Config = constellation): NonEmptyList[AWSStorageConfig] = {
+    val accessKey = constellationConfig.getString(s"storage.aws.aws-access-key")
+    val secretKey = constellationConfig.getString(s"storage.aws.aws-secret-key")
+    val region = constellationConfig.getString(s"storage.aws.region")
+
+    def load(bucket: String): AWSStorageConfig =
+      AWSStorageConfig(
+        accessKey = accessKey,
+        secretKey = secretKey,
+        region = region,
+        bucket = bucket
+      )
+
+    val backupStorage: List[String] = Try(
+      constellationConfig.getStringList("storage.aws.backup-buckets-names").asScala.toList
+    ).getOrElse(Nil)
+
+    val base = load(constellationConfig.getString(s"storage.aws.bucket-name"))
+    val backups = backupStorage.map(load)
+
+    NonEmptyList(base, backups)
+  }
```

Comment:
Can we reuse the cloud config which is already prepared? Maybe just an optional flag there in each entry that the entry should be used as read-only. (or write-only for cold backups only)

Reply:
I will check that!

Reply:
Ok I looked into it and I think it would be a nice improvement to not specify the same connection details in two different configs, but to do it well I would need to add handling of different credentials for S3 (Credentials, InheritS3) in SnapshotS3Storage, Info, Rewards and others. Also I would need to handle other providers like GCP similarily and remove the old way of loading providers through application.conf. I think It would be nice to have it, but I would rather not delve into it right now :).